internal/datanode/data_sync_service.go

```diff
@@ -160,6 +161,7 @@ func initMetaCache(initCtx context.Context, storageV2Cache *metacache.StorageV2C
 				if params.Params.CommonCfg.EnableStorageV2.GetAsBool() {
 					stats, err = loadStatsV2(storageV2Cache, segment, info.GetSchema())
 				} else {
+					binlog.DecompressBinLogs(segment)
```

Comment:
don't do DecompressBinLogs here.
We should do it in node.broker.GetSegmentInfo call.

Datanode and Querynode, Querycoord as the consumer of Segment info, should do DecompressBinLogs in the  rpc call itself